components/usage/pkg/controller/reconciler.go

```diff
@@ -30,12 +30,13 @@ func (f ReconcilerFunc) Reconcile() error {
 }
 
 type UsageReconciler struct {
-	nowFunc func() time.Time
-	conn    *gorm.DB
+	nowFunc        func() time.Time
+	paymentEnabled bool
+	conn           *gorm.DB
 }
 
-func NewUsageReconciler(conn *gorm.DB) *UsageReconciler {
-	return &UsageReconciler{conn: conn, nowFunc: time.Now}
+func NewUsageReconciler(conn *gorm.DB, paymentEnabled bool) *UsageReconciler {
```

Comment:
How about injecting a BillingController object, which if payment is disabled has a no-op implementation? That way, we don't need to push the decision logic on what to run with into the controller and can leave it at the instantiation level.